### PR TITLE
support password for Redis cluster

### PR DESCRIPTION
**Pull request by jzh800**
default password is NULL

### CODE REVIEW

**Comment by eyjian** [comment]
sorry: FILE should be __FILE__:
(*g_error_log)("[%s:%d]auth failure\n", __FILE__, __LINE__);

File: r3c.cpp [hunk 1]
```diff
@@ -2305,13 +2320,24 @@ redisContext* CRedisClient::get_redis_context(unsigned int slot, std::pair<std::
...
                 (*g_error_log)("[%s:%d][standalone]redisConnect failed\n", __FILE__, __LINE__, slot);
             }
             else if (_data_timeout_milliseconds > 0)
             {
+				if(_password.size() > 0)
+				{
+					redisReply* redis_reply = (redisReply*)redisCommand(redis_context, "auth %s", _password.c_str());
+					if (NULL == redis_reply)
+					{
+						(*g_error_log)("[%s:%d]auth failure\n", FILE, LINE);
```
Comment (eyjian) [inline r3c.cpp]: FILE => __FILE__
LINE => __LINE__

**Comment by eyjian** [comment]
Both FILE and LINE need double underlines

**Comment by eyjian** [comment]
fix compile warning and password now is the last parameter, please git pull the last edition commited by me.

r3c.h:449:17: 警告：‘r3c::CRedisClient::_nodes_string’将在 [-Wreorder]
     std::string _nodes_string;
                 ^
r3c.h:447:17: 警告：‘std::string r3c::CRedisClient::_password’ [-Wreorder]
     std::string _password;
                 ^
r3c.cpp:482:1: 警告：在此处初始化后被初始化 [-Wreorder]
 CRedisClient::CRedisClient(const std::string& nodes, std::string password, int connect_timeout_milliseconds, int data_timeout_milliseconds) throw (CRedisException)

**Comment by jzh800** [comment]
we only need IP:port and password,
user may no need care this two parameter,
but they must fill these two paramenters and then could using password,
=CONNECT_TIMEOUT_MILLISECONDS, int data_timeout_milliseconds=DATA_TIMEOUT_MILLISECONDS

anther advice, maybe we could modify the readme to support password.

https://redis.io/clients#c--   intrudction could modify will more ok.

**Pull request merged**